src/select.rs

```diff
@@ -6,13 +6,19 @@ use std::ascii::AsciiExt;
 use string_cache::{Atom, Namespace};
 use tree::{NodeRef, NodeData, ElementData};
 
+/// The definition of whitespace per CSS Selectors Level 3 § 4.
+///
+/// Copied from rust-selectors.
+static SELECTOR_WHITESPACE: &'static [char] = &[' ', '\t', '\n', '\r', '\x0C'];
+
+#[derive(Debug)]
 pub struct KuchikiSelectors;
 
 impl SelectorImpl for KuchikiSelectors {
     type NonTSPseudoClass = PseudoClass;
     type PseudoElement = PseudoElement;
```

Comment:
Sorry for the lack of documentation in the selectors crate. It became more generic over the string types used for various bits of a selector. Based on types used in kuchiki::tree, I suggest:

```rust
    type AttrValue = String;
    type Identifier = String;
    type ClassName = String;
    type LocalName = Atom;
    type NamespaceUrl = Namespace;
    type NamespacePrefix: String;
    type BorrowedNamespaceUrl: Namespace;
    type BorrowedLocalName: Atom;
```

where `Atom` and `Namespace` are from the string-cache crate.